sklearn/datasets/tests/test_openml.py

```diff
@@ -147,32 +149,64 @@ def _monkey_patch_webbased_functions(context, data_id, gziped_files):
         path_suffix = '.gz'
         read_fn = gzip.open
 
+    class MockHTTPResponse():
+        def __init__(self, data, is_gzip):
+            self.data = data
+            self.is_gzip = is_gzip
+
+        def read(self, amt=None):
+            return self.data.read(amt)
+
+        def close(self):
+            self.data.close()
+
+        def getheader(self, name, default=None):
+            if self.is_gzip and name == 'Content-Encoding':
+                return 'gzip'
+            return ''
+
     def _file_name(url, suffix):
         return (re.sub(r'\W', '-', url[len("https://openml.org/"):])
                 + suffix + path_suffix)
 
-    def _mock_urlopen_data_description(url):
+    def _mock_urlopen_data_description(url, has_gzip_header):
         assert url.startswith(url_prefix_data_description)
 
         path = os.path.join(currdir, 'data', 'openml', str(data_id),
                             _file_name(url, '.json'))
-        return read_fn(path, 'rb')
 
-    def _mock_urlopen_data_features(url):
-        assert url.startswith(url_prefix_data_features)
+        if has_gzip_header:
+            fp = open(path, 'rb')
+            return MockHTTPResponse(fp, True)
+        else:
+            fp = read_fn(path, 'rb')
+            return MockHTTPResponse(fp, False)
```

Comment:
This branch is not tested, can we add a test for it?

Reply:
This PR has been updated to test both branches.